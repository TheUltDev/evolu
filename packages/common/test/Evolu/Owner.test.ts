import { expect, test } from "vitest";
import { createOwner, createWriteKey } from "../../src/index.js";
import {
  testCreateMnemonic,
  testCreateRandomBytesDep,
  testMnemonic,
  testMnemonicSeed,
  testTime,
} from "../_deps.js";

test("createOwner", () => {
  const owner = createOwner({
    createMnemonic: testCreateMnemonic,
    ...testCreateRandomBytesDep,
    time: testTime,
  })(testMnemonic);
  expect(owner).toMatchSnapshot();
});

test("createWriteKey", () => {
  const key = createWriteKey(testCreateRandomBytesDep)(testMnemonicSeed);
  expect(key).toMatchSnapshot();
});
