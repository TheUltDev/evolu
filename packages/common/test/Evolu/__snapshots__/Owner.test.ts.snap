// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`createOwner 1`] = `
{
  "createdAt": "1970-01-01T00:00:00.003Z",
  "encryptionKey": Uint8Array [
    173,
    165,
    204,
    225,
    74,
    240,
    76,
    99,
    46,
    18,
    104,
    4,
    0,
    136,
    188,
    132,
    99,
    57,
    151,
    4,
    147,
    44,
    59,
    91,
    61,
    162,
    116,
    243,
    186,
    98,
    25,
    4,
  ],
  "id": "MdVYFAxShUluuZKVWQfYL",
  "mnemonic": "knee easy fork attitude drink gloom head latin spider grab spy reason",
  "writeKey": Uint8Array [
    111,
    64,
    118,
    108,
    245,
    65,
    176,
    98,
    173,
    247,
    85,
    78,
    30,
    167,
    192,
    203,
  ],
}
`;

exports[`createWriteKey 1`] = `
Uint8Array [
  111,
  64,
  118,
  108,
  245,
  65,
  176,
  98,
  173,
  247,
  85,
  78,
  30,
  167,
  192,
  203,
]
`;
